hydrosdk/cluster.py

```diff
@@ -1,40 +1,78 @@
+import json
 import logging
 from urllib import parse
-import requests
-import json
+
 import grpc
-import functools
+import requests
 
 
 class Cluster:
     """
     Cluster responsible for server interactions
     """
+
     @staticmethod
-    def connect(address):
+    def connect(http_address: str, grpc_address: str = None) -> 'Cluster':
         """
         The preferable factory method for Cluster creation. Use it.
 
-        :param address: connection address
-        :raises ConnectionError: if no connection with cluster
+        :param http_address: http connection address
+        :param grpc_address: optional grpc connection address
         :return: cluster object
 
         Checks the address, the connectivity, and creates an instance of Cluster.
         """
-        cl = Cluster(address)
-        logging.info("Connecting to {} cluster".format(cl.address))
-        info = cl.build_info()
+        cl = Cluster(http_address=http_address, grpc_address=grpc_address)
+
+        Cluster.check_connection(cluster=cl, address=http_address, address_type="http")
+        if grpc_address:
+            Cluster.check_connection(cluster=cl, address=grpc_address, address_type="grpc")
+
+        return cl
+
+    @staticmethod
+    def check_connection(cluster: 'Cluster', address: str, address_type: str) -> None:
```

Comment:
I think it's kinda weird to create a static method and pass Cluster as first parameter?

Reply:
`check_connection` is used inside staticmethod `connect`. If `check_connection` is a class method then `connect` should be as well?

Reply:
Connect is a factory method. The reason it's static because we don't have an instance of the class yet, and we will create it.

With `Cluster.check_connection` however, it's different. You already created an instance of Cluster, and pass it to the method. That suits the description of a non-static method.
In other words, you created an additional code, which is unnecessary and could achieved by simple methods.

Reply:
Fixed